### ISSUE TITLE
All null JSON fields get masked

### ISSUE THREAD

**Issue by Sam-Kruglov** [zalando/logbook#653]
If some field is `null`, it gets replaced by `XXX` by default. I couldn't find any way to change it other than copy-pasting and tweaking `JsonBodyFilters`.

```
public static void main(String[] args) {
    val result = BodyFilters.defaultValue().filter(MediaType.APPLICATION_JSON_UTF8_VALUE, "{\n" +
            "  \"myRandomField\": null\n" +
            "}");
    System.out.println(result);
}
```
Prints this:
`{"myRandomField":"XXX"}`

**Comment by Sam-Kruglov** [comment]
This looks intentional.
I can "fix" it by removing the last concatenation here (`org.zalando.logbook.json.JsonBodyFilters#replacePrimitiveJsonProperty`):
```
final Pattern pattern = compile("(" + property + "\\s*:\\s*)" + value + "|null");
final Pattern pattern = compile("(" + property + "\\s*:\\s*)" + value);
```

**Comment by Sam-Kruglov** [comment]
For that "fix" I had to copy-paste `JsonBodyFilters` and `JsonMediaType`

**Comment by whiskeysierra** [comment]
Thanks for the issue! That's indeed an undesired effect. I prepared a fix in #654.

My guess why we haven't noticed this earlier is primarily https://opensource.zalando.com/restful-api-guidelines/#123 which advocates to not include properties with null values at all in any JSON document.
